### PR TITLE
Backport java 17 build support to 7.17

### PR DESCRIPTION
**Pull request by masseyke**
Backporting support for building on java 17 from #1802 and #1808

### CODE REVIEW

File: hive/build.gradle [hunk 1]
```diff
@@ -58,6 +58,12 @@ dependencies {
     javadocSources(project(":elasticsearch-hadoop-mr"))
 }
 
+tasks.named("test").configure {
+    onlyIf { JavaVersion.current() >= JavaVersion.VERSION_16 }
```
Comment (jbaiera) [inline hive/build.gradle:62]: Is this just needed to conditionally add the jvmArgs settings or are these tests required to run on JVM 16 and up?

Comment (masseyke) [reply hive/build.gradle:62]: This is just for the jvmArgs settings. The tests still run for all versions right?

Comment (jbaiera) [reply hive/build.gradle:62]: I think the `onlyIf` method marks the task to only run when the condition is true

**Review by jbaiera — APPROVED** [review]
LGTM, thanks!

**Pull request merged**